litmus-portal/authentication/pkg/entities/user.go

```diff
@@ -0,0 +1,67 @@
+package entities
+
+import (
+	"litmus/litmus-portal/authentication/pkg/utils"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+//User contains the user information
+type User struct {
+	ID        primitive.ObjectID `bson:"_id,omitempty" json:"_id,omitempty"`
+	UserName  string             `bson:"username,omitempty" json:"username,omitempty"`
+	Password  string             `bson:"password,omitempty" json:"password,omitempty"`
+	Email     string             `bson:"email,omitempty" json:"email,omitempty"`
+	Name      string             `bson:"name,omitempty" json:"name,omitempty"`
+	Role      Role               `bson:"role,omitempty" json:"role,omitempty"`
+	LoggedIn  bool               `bson:"logged_in,omitempty" json:"logged_in,omitempty"`
+	CreatedAt *time.Time         `bson:"created_at,omitempty" json:"created_at,omitempty"`
+	UpdatedAt *time.Time         `bson:"updated_at,omitempty" json:"updated_at,omitempty"`
+	RemovedAt *time.Time         `bson:"removed_at,omitempty" json:"removed_at,omitempty"`
+}
+
+//UserPassword defines structure for password related requests
+type UserPassword struct {
+	Username    string `json:"username,omitempty"`
```

Comment:
Same as above

Reply:
- The `/reset/password` POST route does not send a `Username`, hence `omitempty` is required